bcipy/simulator/sim_copy_phrase.py

```diff
@@ -0,0 +1,148 @@
+from typing import Optional
+
+import numpy as np
+
+from bcipy.config import SESSION_DATA_FILENAME
+from bcipy.helpers import load
+from bcipy.helpers.list import grouper
+from bcipy.helpers.session import evidence_records, read_session
+from bcipy.helpers.symbols import alphabet
+from bcipy.signal.model import PcaRdaKdeModel
+from bcipy.simulator.helpers.signal_helpers import process_raw_data_for_model, ExtractedExperimentData
+from bcipy.simulator.helpers.sim_viz import plot_replay_comparison
+from bcipy.simulator.simulator_base import Simulator, SimulatorData
+
+
+class SimulatorCopyPhraseReplay(Simulator):
+    """Simulator.
+
+    Base class for BciPy Simulations.
+
+    Requirements:
+    - run closed loop simulation of {TaskType} with {data} with {simulation_params}
+    """
+
+    def __init__(self, parameter_path, save_dir, signal_models: [tuple], language_models, replay_data_path=None,
+                 verbose=False):
+        super(SimulatorCopyPhraseReplay, self).__init__()
+
+        self.parameters = self.load_parameters(parameter_path)
+        self.save_dir = save_dir
+        self.signal_models = signal_models
+        self.language_models = language_models
+        self.replay_data_path = replay_data_path
+
+        self.symbol_set = alphabet()
+        self.write_output = False
+        self.data_loader = None
+
+        self.signal_models_classes = [PcaRdaKdeModel for m in self.signal_models] # Hardcoded rn
+
+    def run(self):
+        self.init_data_loader()
+
+        self.logger.info(f"Initialized data loader")
+
+        # TODO need to support multimodal and language models
+        model_file, model_class = self.signal_models[0], self.signal_models_classes[0] # hardcoded rn
+        model = model_class(self.parameters.get("k_folds"))
+        model = model.load(model_file)
+
+        self.logger.info(f"Loaded model from {model_file}, with model_class {model_class}")
+
+        target_eeg_evidences, non_target_eeg_evidences = [], []
+        while not self.data_loader.is_end():
```

Comment:
As mentioned below, if the data loader produces an iterator, this could be:

    for inquiry_trials, this_inquiry_letters, this_inquiry_labels in self.inquiry_data:
        ...